### DISCUSSION TITLE
[API] Repositories API does not allow setting "Require pull request" branch protection

### DISCUSSION THREAD

**Discussion by bobheadxi** [community/community#8692] [Pull Requests]
**tl;dr** How does one configure a "Require pull request" branch protection through the API?

It seems that there's no explicit option for only requiring a pull request, i.e. the equivalent of the following, in the [v3 API](https://docs.github.com/en/rest/reference/repos#update-branch-protection):

![image](https://user-images.githubusercontent.com/23356519/145657105-c60e17a3-cdc5-4f6f-b2df-4fbe059781cd.png)

When you [get the protections](https://docs.github.com/en/rest/reference/repos#get-branch-protection) for a branch with this configured, you get the following:

```
Restrictions: &{/* all zero values */, RequiredApprovingReviewCount:0}
```

However, attempting to set `RequiredApprovingReviewCount: 0` in an update request returns an error from the API indicating the value must be greater than or equal to 1:

```
    branch_test.go:58: PUT https://api.github.com/repos/sourcegraph/sourcegraph/branches/test-buildsherrif-branch/protection: 422 Invalid request.
        
        No subschema in "anyOf" matched.
        0 must be greater than or equal to 1.
        Not all subschemas of "allOf" matched.
        For 'anyOf/1', {"dismiss_stale_reviews"=>false, "require_code_owner_reviews"=>false, "required_approving_review_count"=>0} is not a null. []
```

**Comment by bobheadxi** [comment]
This works in the GraphQL API, but not in the REST API:

```gql
mutation {
	  updateBranchProtectionRule(input: {
		  branchProtectionRuleId: "foobar",
	  
		  requiresApprovingReviews: true,
		  requiredApprovingReviewCount: 0
	  }) {
		  clientMutationId
	  }
}
```
